### ISSUE TITLE
PNN not serializing properly

### ISSUE THREAD

**Issue by syats** [itdxer/neupy#171]
The following code creates a PNN object and tries to save it to a file using dill, as suggested in[ this doc](http://neupy.com/docs/storage.html). The code throws an exception, so that there is effectively no way to save an already trained PNN. Any suggestions would be appreciated.

```

from neupy.algorithms import PNN
import numpy as np
import dill

#Create data
X = np.zeros((100,10))
#Create class labels (binary)
Y = np.zeros((100,1))
Ybin = np.zeros_like(Y)
Ybin[Y>0.3] = 1

#Create and train PNN
N = PNN()
N.train(X,Y)

#Save PNN
with open('network-storage.dill', 'wb') as f:
    dill.dump(N, f)

```

The exception

```
Traceback (most recent call last):
  File "neupybug.py", line 15, in <module>
    dill.dump(N, f)
  File "/usr/local/lib/python3.4/dist-packages/dill/dill.py", line 252, in dump
    pik.dump(obj)
  File "/usr/lib/python3.4/pickle.py", line 412, in dump
    self.save(obj)
  File "/usr/lib/python3.4/pickle.py", line 524, in save
    self.save_reduce(obj=obj, *rv)
  File "/usr/lib/python3.4/pickle.py", line 627, in save_reduce
    save(state)
  File "/usr/lib/python3.4/pickle.py", line 479, in save
    f(self, obj) # Call unbound method with explicit self
  File "/usr/local/lib/python3.4/dist-packages/dill/dill.py", line 841, in save_module_dict
    StockPickler.save_dict(pickler, obj)
  File "/usr/lib/python3.4/pickle.py", line 814, in save_dict
    self._batch_setitems(obj.items())
  File "/usr/lib/python3.4/pickle.py", line 840, in _batch_setitems
    save(v)
  File "/usr/lib/python3.4/pickle.py", line 499, in save
    rv = reduce(self.proto)
  File "/usr/local/lib/python3.4/dist-packages/neupy/utils.py", line 156, in __getattr__
    return self[attrname]
KeyError: '__getstate__'

```

Using: Python 3.4.2 on Linux [GCC 4.9.1]
and Neupy 0.5.1

**Comment by itdxer** [comment]
Hi
That's a bug and the code suppose to work. I have to fix it later. Thank you for reporting.

PNN use something known as Lazy Learning, which means it doesn't train at all. It just copy input variables and later it use it for prediction, Basically you can use function 'train' everytime you want to use PNN, because it just loads parameters which is technically the same as reading pickled file.

**Comment by itdxer** [comment]
It's fixed in the latest release branch `release/v0.5.2`. This algorithm works fine even with `pickle`

**Comment by syats** [comment]
Excellent! Thanks a lot!
